### ISSUE TITLE
WaitAndRetry action with a count.

### ISSUE THREAD

**Issue by Mark-Broadhurst** [App-vNext/Polly#26]
I'd like to get an error count on the action when I use WaitAndRetry much like the standard Retry so I can avoid writing something like this.

Policy.Handle<Exception>()
            .Retry(3, (exception, retryCount) => 
        {
            if (retryCount < 3)
            {
                logger.Warn("SocialMedia.Checker", exception);
                Thread.Sleep(TimeSpan.FromSeconds(3));
            }
            if (retryCount == 3)
            {
                logger.Error("SocialMedia.Checker", exception);
            }
            Thread.Sleep(TimeSpan.FromMinutes(1));
        });

**Comment by michael-wolfenden** [comment]
Mark,

That functionality does not yet exist, however in the meantime you can achieve something similar by maintaining your own count like below:

```csharp
var retryCount = 0;
		
var policy = Policy
	.Handle<DivideByZeroException>()
	.WaitAndRetry(
		5, 
		retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)), 
		(exception, timeSpan) => {
			retryCount += 1;
			// if retryCount ...
		}
	);
```

**Comment by reisenberger** [comment]
Closed in #110
